assets/src/edit-story/components/panels/design/preset/presetPanel.js

```diff
@@ -57,16 +64,20 @@ function PresetPanel({
   );
 
   const { colors, textStyles } = stylePresets;
-  const presets = isColor ? colors : textStyles;
-  const hasPresets = presets.length > 0;
+  const globalPresets = isColor ? colors : textStyles;
+  const { colors: localColors } = storyPresets;
+  const hasLocalPresets = localColors.length > 0;
+  const hasPresets = isColor
+    ? globalPresets.length > 0 || hasLocalPresets
+    : globalPresets.length > 0;
```

Comment:
This can be simplified. `globalPresets.length > 0 || (isColor && hasLocalPresets)` seems to do the trick, no? (but it could use a comment regardless)